### PR TITLE
Wait for api-server to report version after starting

### PR DESCRIPTION
**Pull request by larmog**
Work-around issue #970 by waiting for apiserver to be up and running before continuing starting the rest of k3s.

### CODE REVIEW

File: pkg/daemons/control/server.go [hunk 1]
```diff
@@ -107,6 +114,42 @@ func Server(ctx context.Context, cfg *config.Control) error {
...
+			logrus.Warnf("apiserver is not yet up and running, will wait one second until next check")
+			time.Sleep(1 * time.Second)
+		} else {
+			logrus.Infof("apiserver %s is up and running", info)
+			break
+		}
+	}
+
+	return nil
+}
```
Comment (zimme) [inline pkg/daemons/control/server.go]: What if we've been waiting 60 seconds, shouldn't it error out then? Now it seems to just continue as if the api server was up and running even though we haven't gotten a response? Did I miss something?

Comment (larmog) [reply pkg/daemons/control/server.go]: You're right, I'll fix that. I'm not sure this is the best way to solve this.

Comment (zimme) [reply pkg/daemons/control/server.go]: Now the code always fail as even if we enter the else statement we `return err` at the end.
If we do `return nil` in the else statement it shouldn't return an error if everything is good.

Comment (larmog) [reply pkg/daemons/control/server.go]: `err` at the end is `nil` after a successful call to the `apiserver` but maybe that's not clear enough. Better just to return `nil` in the `else`. I'll fix

Comment (zimme) [reply pkg/daemons/control/server.go]: Aah, missed that kinda used to early out code and don't work with go much at all :p

**Comment by erikwilson** [comment]
Thanks for the PR @larmog!

**Comment by pcastellazzi** [comment]
Any chance to get this patch on a "quick release"? ... v0.10.3 may be?

**Pull request merged**